plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchQueryBuilder.java

```diff
@@ -54,6 +67,65 @@
 {
     private ElasticsearchQueryBuilder() {}
 
+    private static final Map<String, BiFunction<String, String, AggregationBuilder>> converters =
+            ImmutableMap.of(MetricAggregation.AGG_MAX, (alias, field) -> new MaxAggregationBuilder(alias).field(field),
+                    MetricAggregation.AGG_MIN, (alias, field) -> new MinAggregationBuilder(alias).field(field),
+                    MetricAggregation.AGG_SUM, (alias, field) -> new SumAggregationBuilder(alias).field(field),
+                    MetricAggregation.AGG_AVG, (alias, field) -> new AvgAggregationBuilder(alias).field(field),
+                    MetricAggregation.AGG_COUNT, (alias, field) -> new ValueCountAggregationBuilder(alias, null).field(field));
+
+    private static AggregationBuilder buildMetricAggregation(MetricAggregation aggregation)
+    {
+        Optional<ElasticsearchColumnHandle> column = aggregation.getColumnHandle();
+        String field;
+        if (column.isEmpty()) {
+            field = "_id"; // use value_count("_id") aggregation to resolve count(*)
+        }
+        else {
+            field = column.get().getName();
+        }
+        return converters.get(aggregation.getFunctionName()).apply(aggregation.getAlias(), field);
+    }
+
+    public static List<AggregationBuilder> buildAggregationQuery(
+            List<TermAggregation> termAggregations,
+            List<MetricAggregation> aggregates)
+    {
+        ImmutableList.Builder<AggregationBuilder> aggregationsBuilder = ImmutableList.builder();
+        AggregationBuilder bucketAggregationBuilder = null;
+        TermsAggregationBuilder inner = null;
+        if (termAggregations != null && !termAggregations.isEmpty()) {
+            for (TermAggregation termAggregation : termAggregations) {
+                TermsAggregationBuilder tab = new TermsAggregationBuilder(termAggregation.getTerm(), null)
+                        .field(termAggregation.getTerm())
+                        .size(10000); // TODO: make the bucket count configurable
```

Comment:
Good idea! Composite aggregations can support pagination. I will work on it, but should take more time, since many code should be rewrited

Reply:
@aalbu Composite aggregation is introduced in es >= 6.1, do we need to check the version and fall back to nested term aggregation when es version not above 6.1?

Reply:
I would suggest we don't push down for versions < 6.1.  Having two ways of pushing down aggregation doesn't seem justified, IMO (especially since we're not excluding a lot of the supported versions).

Reply:
@aalbu another version problem i found.  That is about the result returned by es.
6.1/6.2: no `after_key`, just use last bucket as the next retrieve parameter
>= 6.3: the result contains `after_key`, use `after_key` as the next retrieve, and not suggest using last bucket as 6.1/6.2 behavior
do we need to consider the difference?

Reply:
Pretty annoying.  I would say let's just support aggregate push-down for versions >= 6.3 to keep it simple.  I don't have a good sense for what our users are deploying, but I would hope most are already upgraded to 7.x.  @martint any thoughts on that?